frontend/src/metabase-lib/parameters/utils/click-behavior.ts

```diff
@@ -0,0 +1,330 @@
+import {
+  dimensionFilterForParameter,
+  variableFilterForParameter,
+} from "metabase-lib/parameters/utils/filters";
+import type { Dimension } from "metabase-lib/types";
+import { isa, isDate } from "metabase-lib/types/utils/isa";
+import { TYPE } from "metabase-lib/types/constants";
+import TemplateTagVariable from "metabase-lib/variables/TemplateTagVariable";
+import { TemplateTagDimension } from "metabase-lib/Dimension";
+import type Question from "metabase-lib/Question";
+import type { ClickObjectDataRow } from "metabase-lib/queries/drills/types";
+
+interface SourceFilters {
+  column: (column: DatasetColumn) => boolean;
+  parameter: (parameter: Parameter) => boolean;
+  userAttribute: (userAttribute: string) => boolean;
+}
+
+interface ParameterTarget {
+  type: "parameter";
+  id: Parameter["id"];
+}
+
+interface Target {
+  id: Parameter["id"];
+  name: Parameter["name"];
+  target: ParameterTarget;
+  sourceFilters: SourceFilters;
+}
+
+export function getDataFromClicked({
+  extraData: { dashboard, parameterValuesBySlug = {}, userAttributes } = {},
+  dimensions = [],
+  data = [],
+}: {
+  extraData?: {
+    dashboard?: Dashboard;
+    parameterValuesBySlug?: Record<string, ParameterValueOrArray>;
+    userAttributes?: UserAttribute[] | null;
+  };
+  dimensions: Dimension[];
+  data: (ClickObjectDataRow & {
+    clickBehaviorValue: ClickObjectDataRow["value"];
+  })[];
+}): ValueAndColumnForColumnNameDate {
+  const column = [
+    ...dimensions,
+    ...data.map(d => ({
+      column: d.col,
+      // When the data is changed to a display value for use in tooltips, we can set clickBehaviorValue to the raw value for filtering.
+      value: d.clickBehaviorValue || d.value,
+    })),
+  ]
+    .filter(d => d.column != null)
+    .reduce<ValueAndColumnForColumnNameDate["column"]>(
+      (acc, { column, value }) => {
+        if (!column) {
+          return acc;
+        }
+
+        const name = column.name.toLowerCase();
+
+        if (acc[name] === undefined) {
+          return { ...acc, [name]: { value, column } };
+        }
+
+        return acc;
+      },
+      {},
+    );
+
+  const dashboardParameters = (dashboard?.parameters || []).filter(
+    ({ slug }) => parameterValuesBySlug[slug] != null,
+  );
+
+  const parameterByName =
+    dashboard == null
+      ? {}
+      : Object.fromEntries(
+          dashboardParameters.map(({ name, slug }) => [
+            name.toLowerCase(),
+            { value: parameterValuesBySlug[slug] },
+          ]),
+        );
+
+  const parameterBySlug = _.mapObject(parameterValuesBySlug, value => ({
+    value,
+  }));
+
+  const parameter =
+    dashboard == null
+      ? {}
+      : Object.fromEntries(
+          dashboardParameters.map(({ id, slug }) => [
+            id,
+            { value: parameterValuesBySlug[slug] },
+          ]),
+        );
+
+  const userAttribute = Object.fromEntries(
+    (userAttributes || []).map(value => [value, { value }]),
```

Comment:
`userAttributes` is an array, but previously it was assumed that it's an object.

Reply:
When setting the click behavior to link to a dashboard I still get `userAttributes` that's an object.
![image](https://github.com/metabase/metabase/assets/1937582/2b723d74-2de9-447a-b91d-bc055caf43cc)

This prompts a console warning
```
Visualization.jsx:253 TypeError: (userAttributes || []).map is not a function
    at getDataFromClicked (click-behavior.ts:118:1)
    at getClickBehaviorData (dashboard-click-drill.js:132:1)
    at getDashboardDrillType (dashboard-click-drill.js:21:1)
    at DashboardClickAction (DashboardClickAction.tsx:67:1)
    at Mode.ts:86:1
    at Array.flatMap (<anonymous>)
    at Mode.actionsForClick (Mode.ts:85:1)
    at Visualization.getClickActions (Visualization.jsx:237:1)
    at Visualization.jsx:251:1
    at TableSimple.jsx:90:1
```

I know there are other `userAttributes` that's fetched from `GET /api/mt/user/attributes` that will return the list of all available user attributes in an array

https://github.com/metabase/metabase/blob/07022b1eeda80f473ad4b60ebbb80187ef85acdf/docs/api/ee/sandbox-user.md#L11-L14

I'm not sure we got the type mixed up somehow. I think we need to assume this still is an object.

Reply:
Great catch!

Fixed with 935ea384a800f394da9dfb7b0fb2f0b6c34d6555